scenario/state.py

```diff
@@ -616,14 +618,104 @@ def __repr__(self):
 
 def _status_to_entitystatus(obj: StatusBase) -> _EntityStatus:
     """Convert StatusBase to _EntityStatus."""
-    statusbase_subclass = type(StatusBase.from_name(obj.name, obj.message))
+    return {
+        ops.UnknownStatus: UnknownStatus,
+        ops.ErrorStatus: ErrorStatus,
+        ops.ActiveStatus: ActiveStatus,
+        ops.BlockedStatus: BlockedStatus,
+        ops.MaintenanceStatus: MaintenanceStatus,
+        ops.WaitingStatus: WaitingStatus,
+    }[obj.__class__](obj.message)
+
+
+@dataclasses.dataclass(frozen=True, eq=False, repr=False)
+class UnknownStatus(_EntityStatus, ops.UnknownStatus):
+    """The unit status is unknown.
+
+    A unit-agent has finished calling install, config-changed, and start, but the
+    charm has not called status-set yet.
+    """
+
+    name: Literal["unknown"] = "unknown"
+
+    def __init__(self):
+        super().__init__(name=self.name)
+
+
+@dataclasses.dataclass(frozen=True, eq=False, repr=False)
+class ErrorStatus(_EntityStatus, ops.ErrorStatus):
+    """The unit status is error.
+
+    The unit-agent has encountered an error (the application or unit requires
+    human intervention in order to operate correctly).
+    """
+
+    name: Literal["error"] = "error"
+
+    def __init__(self, message: str = ""):
+        super().__init__(name="error", message=message)
+
+
+@dataclasses.dataclass(frozen=True, eq=False, repr=False)
+class ActiveStatus(_EntityStatus, ops.ActiveStatus):
+    """The unit/app is ready.
+
+    Use the :attr:`message` to provide details when the unit/app is operational
+    but in a degraded state (such as lacking high availability).
+    """
+
+    name: Literal["active"] = "active"
+
+    def __init__(self, message: str = ""):
+        super().__init__(name="active", message=message)
+
+
+@dataclasses.dataclass(frozen=True, eq=False, repr=False)
+class BlockedStatus(_EntityStatus, ops.BlockedStatus):
+    """The unit/app requires manual intervention.
```

Comment:
Yeah, I wondered about that (similar to what Harness does). These are more-or-less what the ops ones will be once the [upcoming changes](https://github.com/canonical/operator/pull/1148#issuecomment-2100607689) make it in; i just left off the bit about being read-only since it didn't seem relevant.

There are probably a couple of others that could make use of this.